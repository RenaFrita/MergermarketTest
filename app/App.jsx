import React from 'react';
import companies from '../data/company';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.companies = companies;
  }

  render() {
    return (
      <div>
      </div>
    );
  }
}
