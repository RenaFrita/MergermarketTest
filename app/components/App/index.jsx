import React from 'react';
import companies from '../../../data/company';
import Search from '../Search/index.jsx';
import Companies from '../Companies/index.jsx';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      companies: companies,
      queryString: '',
      results: [],
      view: []
    }
    this.query = this.query.bind(this);
    this.reset = this.reset.bind(this);
    this.show = this.show.bind(this);
  }

  query(evt) {
    if (evt.target.value.length > 2) {
      const regx = new RegExp(evt.target.value, 'gi');
      const results = this.state.companies.filter((company) => {
        return company.name.match(regx) !== null;
      });
      this.setState({
        results: results.slice(5, results.length),
        view: results.slice(0, 5),
        queryString: evt.target.value
      });
    } else {
      this.setState({ results: [], view: [], queryString: evt.target.value });
    }
  }

  reset() {
    this.setState({ results: [], view: [], queryString: '' });
  }

  show() {
    if (this.state.results.length) {
      this.setState({
        results: this.state.results.slice(5, this.state.results.length),
        view: this.state.view.concat(this.state.results.slice(0, 5))
      });
    }
  }

  render() {
    return (
      <div>
        <Search
          query={this.query}
          reset={this.reset}
          queryString={this.state.queryString}
        />
        <Companies
          companies={this.state.view}
          show={this.state.results.length ? this.show : undefined}
        />
      </div>
    );
  }
}
