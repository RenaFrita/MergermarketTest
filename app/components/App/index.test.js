import React from 'react';
import App from './index.jsx';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('App component', () => {
  test('should render the correct markup', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  const app = mount(<App />);
  test('should display 5 results', () => {
    app.find('input').simulate('change', {target: {value: 'son'}});
    expect(app.state(['view']).length).toBe(5);
  });

  test('should display 10 results', () => {
    app.find('.companies-more').simulate('click');
    expect(app.state(['view']).length).toBe(10);
  });

  test('should display 0 results', () => {
    app.find('.search-clear').simulate('click');
    expect(app.state(['view']).length).toBe(0);
  });
});
