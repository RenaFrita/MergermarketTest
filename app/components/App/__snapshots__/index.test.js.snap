// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render the correct markup 1`] = `
<div>
  <form
    className="search-form"
  >
    <label
      className="search-label"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test-file-stub",
        }
      }
      htmlFor="search"
    />
    <input
      className="search-input"
      id="search"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="search-clear"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test-file-stub",
        }
      }
      onClick={[Function]}
      type="button"
    />
  </form>
  <ul
    className="companies"
  >
    <li
      className="companies-headers"
    >
      <span>
        Name
      </span>
      <span>
        Type
      </span>
      <span>
        Headquarter
      </span>
    </li>
    
  </ul>
</div>
`;
